docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,282 @@
+
+## Overview
+
+The Grafana Mimir configuration can be managed through the Helm chart or supplied via a user managed object.
+
+If you want to manage the configuration via the Helm chart, see [Manage the configuration with Helm](#manage-the-configuration-with-helm).
+
+If you want to manage the configuration directly, see [Manage the configuration directly](#manage-the-configuration-directly).
+
+Handling sensitive information, such as credentials is common between the two methods, see [Injecting credentials](#injecting-credentials).
+
+## Manage the configuration with Helm
+
+There are three ways configuration parameters can be modified:
+
+1. Copying the whole `mimir.config` value and modifying the configuration as text
+1. Setting parameters via the `mimir.structuredConfig` value (recommended)
+1. Setting extra CLI flags for components individually
+
+See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
+
+> **Limitation:**: it is not possible to delete configuration parameter settings via `mimir.structuredConfig` that were set in `mimir.config`. Set the configuration parameter to its default or to some other value instead.
+
+### How the configuration is applied
+
+Grafana Mimir components are run with a configuration calculcated by the following process:
+
+1. The configuration YAML in `mimir.config` is evaulated for Helm templates. This step is intended to ensure that the configuration applies to the Kubernetes cluster where it will be installed. For example setting up cluster specific addresses.
+1. The values from `mimir.structuredConfig` are merged on top and the result is again evaulated for Helm templates. This steps is intended to apply user specific customizations. For example S3 storage details.
+1. The resulting YAML configuration is then sorted alphabetically and stored in a `ConfigMap` (or `Secret` depending on the value of `configStorageType`) and provided to all Grafana Mimir components.
+1. When Grafana Mimir components are run in pods, the configuration file as well as any extra CLI flags are provided to the component.
+1. Each component evaulates the configuration, substituting environment variables as required. Note that extra CLI flags take precedence over the configuration file.
+
+> **Note:**: CLI flags are component specific, thus they will not show up in the generated `ConfigMap` (or `Secret`), making it less obvious what configuration is running. Use only when absolutely necessary.
+
+### Inspect changes to the configuration before upgrade
+
+Follow these steps to inspect what change will be applied to the configuration.
+
+Preparation:
+
+1. Install the [helm diff](https://github.com/databus23/helm-diff) plugin.
+1. Make sure to use `configStorageType` is set to `ConfigMap`
```

Comment:
```suggestion
1. Use the default value of `configStorageType: ConfigMap`
```

or something along those lines

Reply:
```suggestion
1. Verify that `configStorageType` value is set to `ConfigMap`.
```

Reply:
how about this ^ ?

Reply:
I wanted to mention that it's also the default. Kind of takes away the anxiety when reading that sentence. Because it's less work than actually needed. I am overthinking this, so your version sounds good too. 👍